src/alert/alert.ts

```diff
@@ -85,7 +91,12 @@ export class NgbDismissibleAlert implements OnInit {
       this.closeEvent.emit($event);
       this.close();
     });
+    if (this.dismissOnTimeout) {
+      setTimeout(() => { this.close(); }, toInteger(this.dismissOnTimeout));
+    }
   }
+
+  ngOnDestroy() { clearInterval(this.dismissOnTimeout); }
```

Comment:
I think you need to store the result of the setTimeout() and use this with clearTimeout() in the ngOnDestroy to correctly remove the dismiss feature